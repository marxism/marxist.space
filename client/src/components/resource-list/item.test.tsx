/* eslint-env jest */

import * as React from 'react'
import Item from './item'
import { create } from 'react-test-renderer'

describe('Item', () => {
  test('renders', () => {
    const props = {
      href: 'https://example.com',
      title: 'foo',
      tags: ['one', 'two'],
      subtitle: 'asdf',
      excerpts: ['one', 'two', 'three'],
    }
    expect(create(<Item {...props} />).toJSON()).toMatchSnapshot()
  })
})
