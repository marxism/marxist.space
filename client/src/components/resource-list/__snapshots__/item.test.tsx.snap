// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 16px;
  line-height: 21px;
  font-weight: 500;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: color 0.1s ease;
  transition: color 0.1s ease;
  text-underline-position: under;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 small {
  font-size: 12px;
}

.c3 {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: 800px;
  padding-bottom: 1px;
  padding-left: 8px;
  font-size: 13px;
  line-height: 19px;
}

.c4 {
  font-size: 12px;
  margin-top: auto;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-left: 1px solid;
  padding: 8px;
  min-width: 0;
}

.c0:not(:first-child) {
  border-top: 1px solid;
}

<li
  className="c0"
>
  <div
    className="c1"
  >
    <a
      className="c2"
      href="https://example.com"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      foo
      <small>
        asdf
      </small>
    </a>
    <div
      className="c3"
    >
      one
    </div>
    <div
      className="c3"
    >
      two
    </div>
    <div
      className="c3"
    >
      three
    </div>
    <span
      className="c4"
    >
      Tags: 
      one two
    </span>
  </div>
</li>
`;
